src/runtime/pythonengine.cs

```diff
@@ -586,7 +586,10 @@ internal static PyObject RunString(string code, IntPtr? globals, IntPtr? locals,
                     code, (IntPtr)flag, globals.Value, locals.Value
                 );
 
-                Runtime.CheckExceptionOccurred();
+                if (result == IntPtr.Zero)
+                {
+                    throw new PythonException();
```

Comment:
What is the significance of this change?

Reply:
`CheckExceptionOccurred` take a P/Invoke calling which means take much overhead, I already test it before, that's a huge gap for some computing intensive code.
Besides, some C function may return `nullptr` without setting a error, we should use the return value as we can.
We can just some helpers methods like that
https://github.com/amos402/pythonnet/commit/3591d92ff0597846cdaa23685d56628e3079ceb0#diff-d5e22b611cf0ccb5d1176bd604442282